DigitalLearningSolutions.Web.Tests/Controllers/Register/RegisterAtNewCentreControllerTests.cs

```diff
@@ -0,0 +1,319 @@
+        private ICentresDataService centresDataService = null!;
+        private RegisterAtNewCentreController controller = null!;
+        private IFeatureManager featureManager = null!;
+
+        private PromptsService promptsService = null!;
+        private IRegistrationService registrationService = null!;
+        private HttpRequest request = null!;
+        private ISupervisorDelegateService supervisorDelegateService = null!;
+        private IUserService userService = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            centresDataService = A.Fake<ICentresDataService>();
+            registrationService = A.Fake<IRegistrationService>();
+            userService = A.Fake<IUserService>();
+            promptsService = A.Fake<PromptsService>();
+            featureManager = A.Fake<IFeatureManager>();
+            supervisorDelegateService = A.Fake<ISupervisorDelegateService>();
+            request = A.Fake<HttpRequest>();
+
+            controller = new RegisterAtNewCentreController(
+                    centresDataService,
+                    featureManager,
+                    promptsService,
+                    registrationService,
+                    supervisorDelegateService,
+                    userService
+                )
+                .WithDefaultContext()
+                .WithMockRequestContext(request)
+                .WithMockServices()
+                .WithMockTempData()
+                .WithMockUser(true);
+        }
+
+        [Test]
+        public void PersonalInformationPost_with_invalid_emails_fails_validation()
+        {
+            // Given
+            const int centreId = 3;
+            controller.TempData.Set(new InternalDelegateRegistrationData());
+            var userAccount = UserTestHelper.GetDefaultUserAccount();
+            var delegateAccount = UserTestHelper.GetDefaultDelegateAccount();
+            var model = new InternalPersonalInformationViewModel
+            {
+                Centre = centreId,
+                CentreSpecificEmail = "centre email",
+            };
+            A.CallTo(() => userService.EmailIsInUse(model.CentreSpecificEmail!))
+                .Returns(true);
+            A.CallTo(() => userService.GetUserById(userAccount.Id)).Returns(
+                new UserEntity(userAccount, new List<AdminAccount>(), new[] { delegateAccount })
+            );
+
+            // When
+            var result = controller.PersonalInformation(model);
+
+            // Then
+            A.CallTo(() => userService.EmailIsInUse(model.CentreSpecificEmail!))
+                .MustHaveHappened();
+            result.Should().BeViewResult().WithDefaultViewName();
+        }
+
+        [Test]
+        public void PersonalInformationPost_with_valid_emails_is_allowed()
+        {
+            // Given
+            controller.TempData.Set(new InternalDelegateRegistrationData());
+            var duplicateUser = UserTestHelper.GetDefaultDelegateUser();

```

Comment:
Why do we need `duplicateUser`? Can't we just specify a random centreId?

Reply:
We just need to specify that it doesn't match the ControllerContextHelper.CentreId, so I've changed the following line to check that.